petab/core.py

```diff
@@ -239,25 +243,27 @@ def get_simulation_to_optimization_parameter_mapping(
         name: idx for idx, name in enumerate(sbml_parameter_ids)
     }
 
-    def _apply_overrides(overrides, condition_id, observable_id):
-        """Apply parameter-overrides to mapping matrix"""
+    def _apply_overrides(overrides, condition_id, observable_id, override_type):
+        """Apply parameter-overrides to mapping matrix
+        override_type: observable / noise
+        """
         for i, override in enumerate(overrides):
-            if isinstance(override, float):
+            if isinstance(override, numbers.Number):
```

Comment:
good. I had the issue multiple times that numpy floats are no ordinary floats, and 64/32 is again different. as far as I understand, numbers.Number should catch all of these.